javaslang/src/test/java/javaslang/collection/euler/Euler43Test.java

```diff
@@ -54,25 +54,22 @@ public void shouldSolveProblem43() {
     private static Seq<Long> tenDigitPandigitalsWithProperty() {
         final CharSeq ALL_DIGITS = CharSeq.of("0123456789");
         final List<Integer> DIVISORS = List.of(2, 3, 5, 7, 11, 13, 17);
-        final char DUMMY_CHAR = Character.MIN_VALUE;
 
         return ALL_DIGITS
                 .combinations(2)
                 .flatMap(CharSeq::permutations)
-                .map(cs -> Tuple.of(cs, cs.prepend(DUMMY_CHAR), Integer.valueOf(cs.mkString())))
                 .flatMap(firstTwoDigits -> DIVISORS
                         .foldLeft(List.of(firstTwoDigits), (accumulator, divisor) -> accumulator
                                 .flatMap(digitsSoFar -> ALL_DIGITS
-                                        .removeAll(digitsSoFar._1)
-                                        .map(nextDigit -> Tuple.of(digitsSoFar._1.append(nextDigit), digitsSoFar._2.tail().append(nextDigit)))
-                                        .map(digitsToTest -> Tuple.of(digitsToTest._1, digitsToTest._2, Integer.valueOf(digitsToTest._2.mkString())))
+                                        .removeAll(digitsSoFar)
+                                        .map(nextDigit -> digitsSoFar.append(nextDigit))
                                 )
-                                .filter(digitsToTest -> digitsToTest._3 % divisor == 0)
+                                .filter(digitsToTest -> Integer.valueOf(digitsToTest.takeRight(3).mkString()) % divisor == 0)
```

Comment:
That would be nice. It is pretty common doing that type of conversions in
the Euler problems.

On 21 August 2016 at 09:33, Daniel Dietrich <notifications@github.com>
wrote:

> In javaslang/src/test/java/javaslang/collection/euler/Euler43Test.java
> <https://github.com/javaslang/javaslang/pull/1514#discussion_r75590213>:
>
> >                                  )
> > -                                .filter(digitsToTest -> digitsToTest._3 % divisor == 0)
> > +                                .filter(digitsToTest -> Integer.valueOf(digitsToTest.takeRight(3).mkString()) % divisor == 0)
>
> I think about adding some syntactic sugar/numeric conversion methods to
> CharSeq, e.g.
>
> Integer.valueOf(digitsToTest.take(3).mkString())
>
> could be
>
> digitsToTest.take(3).toInt()
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/javaslang/javaslang/pull/1514/files/3e3032fae50687edf43e07a348c26d709b0c77d6#r75590213>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADRVwIekHLI2kT3NRnvbQEPDLKQkViWjks5qh_9hgaJpZM4JpO4g>
> .
>